File: code/demonstration/receiver/AndroidIteration3/app/src/main/java/de/fzi/sensidl/receiver/android/iteration3/sensor/HumiditySensorUtil.java
package de.fzi.sensidl.receiver.android.iteration3.sensor;

public class HumiditySensorUtil {

	/**
	 * Convert the humidity
	 * Calculation from the TI Sensor Tag Wiki: http://processors.wiki.ti.com/index.php/SensorTag_User_Guide
	 * @param value the raw value
	 * @return humidity
	 */
	public static float convertHumidity(byte[] value) {
		int a = Util.shortUnsignedAtOffset(value, 2);
		a = a - (a % 4);
		return (-6f) + 125f * (a / 65535f);
	}
}

File: code/demonstration/receiver/XamarinIteration3/XamarinStufe3/XamarinStufe3/Util/HumiditySensorUtil.cs
using System;

namespace XamarinIteration3
{
	public static class HumiditySensorUtil
	{
		// <summary>
		// Convert the humidity
		// Calculation from the TI Sensor Tag Wiki: http://processors.wiki.ti.com/index.php/SensorTag_User_Guide
		// </summary>
		// <param name="value">the raw value</param>
		// <returns>the humidity</returns>
		public static double convertHumidity (byte[] value)
		{
			int a = BitConverter.ToUInt16 (value, 2);
			a = a - (a % 4);
			return (-6f) + 125f * (a / 65535f);
		}
	}
}

